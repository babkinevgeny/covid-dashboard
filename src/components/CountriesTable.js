import React from 'react';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import CountryRow from './CountryRow';
import sortArray from '../helpers/helpers';

class CountriesTable extends React.Component {
  constructor(props) {
    super(props);
    ({ data: this.items, field: this.field } = this.props);
  }

  getRowsData() {
    sortArray(this.items, this.field, false);
    return this.items.map((row) => (
      <tr key={row.Country}>
        <CountryRow
          country={row.Country}
          number={row[this.field]}
        />
      </tr>
    ));
  }

  getKeys() {
    return Object.keys(this.items[0]);
  }

  render() {
    sortArray(this.items, this.field, false);
    return (
      <TableContainer component={Paper}>
        <Table>
          <TableBody>
            {this.items.map((row) => (
              <TableRow key={row.Country}>
                <TableCell>{row.Country}</TableCell>
                <TableCell align="right">{row[this.field]}</TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    // <table>
    //   <tbody>
    //     {this.getRowsData()}
    //   </tbody>
    // </table>
    );
  }
}

// CountriesTable.propTypes = {
//   data: PropTypes.arrayOf(PropTypes.object).isRequired,
// };

export default CountriesTable;
