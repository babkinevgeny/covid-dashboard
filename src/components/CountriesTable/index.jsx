import React from 'react';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableContainer from '@material-ui/core/TableContainer';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import PropTypes from 'prop-types';
import { sortArray } from '../../helpers';

function CountriesTable({ data, field }) {
  const sortedData = sortArray(data, field);

  return (
    <TableContainer component={Paper}>
      <Table>
        <TableBody>
          {sortedData.map((row) => (
            <TableRow key={row.Country}>
              <TableCell>{row.Country}</TableCell>
              <TableCell align="right">{row[field]}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
}

CountriesTable.propTypes = {
  data: PropTypes.arrayOf(PropTypes.object).isRequired,
  field: PropTypes.string.isRequired,
};

export default CountriesTable;
