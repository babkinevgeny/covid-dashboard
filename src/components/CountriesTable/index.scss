.countries_table_row {
  cursor: pointer;

  &:hover {
    background-color: rgba(0, 0, 0, 0.07); ;
  }
}