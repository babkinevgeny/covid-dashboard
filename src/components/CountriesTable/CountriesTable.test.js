import { shallow } from 'enzyme';
import CountriesTable from "../CountriesTable";

const MIN_PROPS = {
  data: [
    {
      Country: 'Country1',
      TestField: 123,
    },
    {
      Country: 'Country2',
      TestField: 321,
    },
  ],
  field: 'TestField',
  newCountryOnRowCLickHandler: jest.fn(),
};

describe('CountriesTable functional component testing', () => {
  
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CountriesTable {...MIN_PROPS} />);
  });

  it('CountriesTable must have a correct snapshot', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('CountriesTable data are sorted', () => {
    const firstRow = wrapper.find('.countries_table_row').at(0);
    expect(firstRow.childAt(0).text()).toBe('Country2');
    expect(firstRow.childAt(1).text()).toBe('321');
  });

});