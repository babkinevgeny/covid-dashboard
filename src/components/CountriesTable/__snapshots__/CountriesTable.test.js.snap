// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountriesTable functional component testing CountriesTable must have a correct snapshot 1`] = `
"<WithStyles(ForwardRef(TableContainer)) component={{...}}>
  <WithStyles(ForwardRef(Table))>
    <WithStyles(ForwardRef(TableBody))>
      <WithStyles(ForwardRef(TableRow)) className=\\"countries_table_row\\" onClick={[Function: onRowClickHandler]}>
        <WithStyles(ForwardRef(TableCell))>
          Country2
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell)) align=\\"right\\">
          321
        </WithStyles(ForwardRef(TableCell))>
      </WithStyles(ForwardRef(TableRow))>
      <WithStyles(ForwardRef(TableRow)) className=\\"countries_table_row\\" onClick={[Function: onRowClickHandler]}>
        <WithStyles(ForwardRef(TableCell))>
          Country1
        </WithStyles(ForwardRef(TableCell))>
        <WithStyles(ForwardRef(TableCell)) align=\\"right\\">
          123
        </WithStyles(ForwardRef(TableCell))>
      </WithStyles(ForwardRef(TableRow))>
    </WithStyles(ForwardRef(TableBody))>
  </WithStyles(ForwardRef(Table))>
</WithStyles(ForwardRef(TableContainer))>"
`;
